File: 06.5_salary_increase_by_category_with_match/java/src/Main.java
import java.io.IOException;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.List;


public class Main {
    private static final List<String> CATEGORIES = Arrays.asList("a","b","c");

    public static void main(String[] args) throws IOException {
        BufferedReader input = new BufferedReader(new InputStreamReader(System.in));

        System.out.println("\n--- PROGRAMA DE AUMENTO DE SALARIO POR CATEGORIAS ---\n");

        System.out.println("Ingresa la categoria a la que perteneces (A, B, C):");
        String category = input.readLine();

        double increment_percentage = 0;

        try {
            increment_percentage = switch (category) {
                case "a", "A" -> 0.15;
                case "b", "B" -> 0.1;
                case "c", "C" -> 0.07;
                default -> {
                    throw new IllegalArgumentException(String.format("La categoria \"%s\" ingresada no existe", category));
                }
            };
        } catch (IllegalArgumentException e) {
            System.err.println(e.getMessage());
            System.exit(1);
        }

        System.out.println("Ingresa tu salario actual:");
        double salary = Integer.parseInt(input.readLine());

        double increment = salary * increment_percentage;
        double new_salary = salary + increment;

        System.out.println("\nTu aumento será de: $ " + increment);
        System.out.println("Tu nuevo salario será de: $ " + new_salary);
    }
}

File: 06.5_salary_increase_by_category_with_match/python/main.py
import sys

CATEGORIES = ['a', 'b', 'c']

if __name__ == '__main__':
    print("\n--- PROGRAMA DE AUMENTO DE SALARIO POR CATEGORIAS ---\n")

    category = input("Ingresa la categoria a la que perteneces (A, B, C) > ")

    match category.lower():
        case 'a' | 'A':
            increment_percentage = 0.15
        case 'b' | 'B':
            increment_percentage = 0.1
        case 'c' | 'C':
            increment_percentage = 0.07
        case _:
            print(f'La categoria "{category}" ingresada no existe', file=sys.stderr)
            sys.exit(1)

    salary = float(input("Ingresa tu salario actual > "))

    increment = salary * increment_percentage
    new_salary = salary + increment

    print(f"\nTu aumento será de: $ {new_salary}")
    print(f"Tu nuevo salario será de: $ {new_salary}")
